R/findVariables.R

```diff
@@ -0,0 +1,57 @@
+#' Find variables and their paths in a Crunch dataset or folder
+#'
+#' @param x Crunch dataset or variable folder
+#' @param deep FALSE (the default) or TRUE; should subfolders
+#' @param include.hidden FALSE (default) or TRUE, should hidden be included in the result?
+#'
+#' @return Data.frame with one row per Crunch variable and columns \code{alias}, \code{path}, \code{hidden}
+#' @export
+findVariables <- function(x, deep = FALSE, include.hidden = FALSE) {
```

Comment:
I think I'd prefer dropping the `include.hidden` param and if you ever wanted it, you'd have to explicitly do `rbind(findVariables(ds), findVariables(hiddenFolder(ds))`, and make this clear in documentation.

(otherwise needs an `include.private` option as well)

The API & documentation are kind of confusing about this, because it changed since I joined, but hidden variables are always in the hidden folder (before 2021, there could be hidden variables in any folder), so I think I'd prefer having the only indication in the output that it is hidden be the folder name.

Reply:
I can understand the case for not specifically dealing with hidden & private variables here, though for me the convenience of having include.hidden and include.private wins out. So the current version has them, but if you feel strongly about this, I'll drop them.